torch/onnx/_internal/fx/exporter.py

```diff
@@ -12,21 +12,67 @@
 import torch._dynamo
 import torch._ops
 import torch.fx
-
 from torch.onnx import _constants
-
 from torch.onnx._internal import _beartype
 from torch.onnx._internal.fx import frontend, function_dispatcher, options, passes
 from torch.utils import _pytree
 
-# TODO: Separate into individual components.
-# TODO: make_fx lose stack info https://github.com/pytorch/pytorch/issues/90276
+
+@contextlib.contextmanager
+def patch_pytree_huggingface_modeloutput():
+    """
+    Patch 'torch.utils._pytree' to support family of 'ModelOutput' from HuggingFace 'transformers'.
+
+    The source and details of the issue is described at https://github.com/pytorch/pytorch/issues/96386.
+    This patch enables 'torch.utils._pytree' to flatten and unflatten all 'ModelOutput'
+    subclasses defined in HuggingFace 'transformers'. Hence resolving the mismatch between
+    `dynamo` eager traced model outputs and `dynamo.export` fx graph computed outputs.
+
+    FIXME(bowbao): Remove this patch after above issue is resolved for `dynamo.export`.
+    """
+    try:
+        from transformers import modeling_outputs  # type: ignore[import]
+    except ImportError as e:
+        # Do nothing if 'transformers' is not installed.
+        try:
+            yield
+        finally:
+            pass
+        return
+
+    def model_output_flatten(
+        output: modeling_outputs.ModelOutput,
+    ) -> Tuple[List[Any], _pytree.Context]:
+        return list(output.values()), (type(output), list(output.keys()))
+
+    def model_output_unflatten(
+        values: List[Any], context: _pytree.Context
+    ) -> modeling_outputs.ModelOutput:
+        output_type, keys = context
+        return output_type(**dict(zip(keys, values)))
+
+    # All 'ModelOutput' subclasses are defined under module 'modeling_outputs'.
+    named_model_output_classes = inspect.getmembers(
+        modeling_outputs,
+        lambda x: inspect.isclass(x) and issubclass(x, modeling_outputs.ModelOutput),
+    )
+
+    for _, class_type in named_model_output_classes:
+        _pytree._register_pytree_node(
+            class_type, model_output_flatten, model_output_unflatten
+        )
+
+    try:
+        yield
+    finally:
+        for _, class_type in named_model_output_classes:
+            _pytree.SUPPORTED_NODES.pop(class_type)
 
 
 @_beartype.beartype
 def _export(
     module: torch.fx.GraphModule,
-    args,
+    *args,
     **kwargs,
```

Comment:
I suggest we keep `args` and `kwargs` once we leave public apis and in places where other arugments exist

Reply:
Good point. Regardless, this `kwargs` needs to change lol.

Reply:
Changed to options and added a related TODO.